test/metabase/lib/metadata/calculation_test.cljc

```diff
@@ -310,3 +310,71 @@
                           {:name       (:name col)
                            :lib/source :source/implicitly-joinable}))
                 (lib/visible-columns query)))))))
+
+(def cols-fns [lib/visible-columns lib/filterable-columns lib/breakoutable-columns lib/orderable-columns])
+
+(deftest ^:parallel inherited-temporal-unit-stage-propagation-test
+  (let [unit :quarter
+        base (-> meta/metadata-provider
+                 (lib/query (meta/table-metadata :orders))
+                 (lib/aggregate (lib/count)))
+        stage-0-query (lib/breakout base
+                                    (lib/with-temporal-bucket (meta/field-metadata :orders :created-at) unit))
+        stage-0-breakout (first (lib/breakouts stage-0-query))
+        stage-0-returned-breakout-col (first (lib/returned-columns stage-0-query))
+        stage-1-query (-> (lib/append-stage stage-0-query)
+                          (lib/with-fields -1 [(assoc stage-0-returned-breakout-col
+                                                      :lib/source :source/previous-stage)]))
+        stage-2-query (lib/append-stage stage-1-query)]
+    (testing "0th stage `orderable-columns` do not contain inherited-temporal-unit"
+      (is ((complement contains?)
+           (lib/find-matching-column stage-0-breakout (lib/orderable-columns stage-0-query))
+           :inherited-temporal-unit)))
```

Comment:
It could be interesting to know if this is true because the column doesn't have `:inherited-temporal-unit` or because we couldn't find a matching column.